### PR TITLE
Add parameter for pin memory of storage to support other devices.

### PR DESCRIPTION
**Pull request by wbigat**
Fixes #ISSUE_NUMBER

Add parameter for pin memory of storage to support other devices.
In the future, other backends will provide their own allocators to create pin memory.

### CODE REVIEW

File: torch/storage.py [hunk 1]
```diff
@@ -191,10 +191,16 @@ def complex_float(self):
         """Casts this storage to complex float type"""
         return self._to(torch.cfloat)
 
-    def pin_memory(self):
+    def pin_memory(self, pin_memory_device="cuda"):
```
Comment (ezyang) [inline torch/storage.py:194]: Can we just call this `device`? Thanks.

Comment (wbigat) [reply torch/storage.py]: OK，this modification has been done. Please take a moment to review it, thanks very much.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**